test/spec/modules/prebidServerBidAdapter_spec.js

```diff
@@ -424,6 +424,58 @@ describe('S2S Adapter', () => {
       $$PREBID_GLOBAL$$.requestBids.removeHook(requestBidsHook);
     });
 
+    it('check gdpr info gets added into cookie_sync request', () => {
+      let cookieSyncConfig = utils.deepClone(CONFIG);
+      cookieSyncConfig.syncEndpoint = 'https://prebid.adnxs.com/pbs/v1/cookie_sync';
+
+      let consentConfig = { consentManagement: { cmpApi: 'iab' }, s2sConfig: cookieSyncConfig };
+      config.setConfig(consentConfig);
+
+      let gdprBidRequest = utils.deepClone(BID_REQUESTS);
+
+      // check scenario if gdprApplies and we got consent information CMP fine
+      gdprBidRequest[0].gdprConsent = {
+        consentString: 'abc123def',
+        gdprApplies: true
+      };
+
+      adapter.callBids(REQUEST, gdprBidRequest, addBidResponse, done, ajax);
+      let requestBid = JSON.parse(requests[0].requestBody);
+
+      expect(requestBid.gdpr).is.equal(1);
+      expect(requestBid.gdpr_consent).is.equal('abc123def');
+
+      // check scenario if gdprApplies is false
+      resetSyncedStatus();
```

Comment:
Tests shouldn't have side-effects on the global state.

It's also dangerous to put this mid-test because it won't execute if one of the earlier assertions fail.

This test should be split into smaller ones, and `resetSyncedStatus()` called in an [afterEach()](https://mochajs.org/#hooks) block

Reply:
True on the above; I wanted to limit the scenarios where I was using this function to not affect other tests by having that function run for every test.  But it doesn't seem like it's breaking the other tests when I put it there now.  I'll make the changes here.

Reply:
it's fine... if other tests fail because the global state gets reset, then _they're_ the broken ones. Tests should be initializing their own state, rather than relying on other tests to do it for them.